application/org.openjdk.jmc.console.agent/src/main/java/org/openjdk/jmc/console/agent/manager/model/PresetRepositoryFactory.java

```diff
@@ -50,9 +50,7 @@
 	public static PresetRepository createSingleton() {
 		if (singleton == null) {
 			synchronized (PresetRepositoryFactory.class) {
-				if (singleton == null) {
-					singleton = create();
-				}
+				singleton = create();
```

Comment:
Actually, singleton field need also to be `volatile`.
see https://shipilev.net/blog/2014/safe-public-construction/ for more info

Reply:
Indeed. I haven't checked, but couldn't all this be avoided by simply create the singleton at the field declaration - does it really have to be lazily created?

Reply:
We could have created a Singleton in field declaration if below points were true :  
1. If we had only "PresetRepository" as a main class with Singleton (Currently this is a factory).
2. Likelihood of "PresetRepository" object required all the time of JMC's lifecycle . 
+ "- does it really have to be lazily created?"
3. Singleton Object + init (PresetRepository.addLocalPresetTo(...) ) is doing a File Operation.

Reply:
> Actually, singleton field need also to be `volatile`.
> see https://shipilev.net/blog/2014/safe-public-construction/ for more info

Thanks @jpbempel that was an informative article and added `volatile` to the field declaration, with this neither Fortify nor apptest complained.